zerver/actions/realm_settings.py

```diff
@@ -104,6 +105,43 @@ def do_set_realm_property(
         update_users_in_full_members_system_group(realm, acting_user=acting_user)
 
 
+@transaction.atomic(durable=True)
+def do_set_realm_permission_group_settings(
+    realm: Realm, name: str, user_group: UserGroup, *, acting_user: Optional[UserProfile]
+) -> None:
+    """Takes in a realm object, the name of an attribute to update, the
+    user_group to update and and the user who initiated the update.
+    """
+
+    old_user_group_id = getattr(realm, name)
+    setattr(realm, name, user_group.id)
+    realm.save(update_fields=[name])
+
+    event = dict(
+        type="realm",
+        op="update",
+        property=name,
+        value=user_group.id,
+    )
```

Comment:
As discussed in https://chat.zulip.org/#narrow/stream/378-api-design/topic/event.20for.20updating.20group.20setting/near/1600758, let's avoid the property/value format here.